src/realm/table.cpp

```diff
@@ -1736,14 +1830,65 @@ void Table::add_search_index(size_t col_ndx)
 }
 
 
-void Table::remove_search_index(size_t col_ndx)
+void Table::remove_search_index(size_t col_ndx, DescriptorRef* subdesc)
 {
     if (REALM_UNLIKELY(!is_attached()))
         throw LogicError(LogicError::detached_accessor);
 
-    if (REALM_UNLIKELY(has_shared_type()))
-        throw LogicError(LogicError::wrong_kind_of_table);
+    if (subdesc) {
+        // Remove index from column of subtable given in subdesc. `this` table must be its root table
+        if (!get_descriptor()->is_root())
+            throw(LogicError::wrong_kind_of_table);
+
+        typedef _impl::DescriptorFriend df;
+
+        if (REALM_UNLIKELY(col_ndx >= subdesc->get()->get_spec()->get_column_count()))
+            throw LogicError(LogicError::column_index_out_of_range);
+
+        // Early-out of non-indexed
+        int attr = subdesc->get()->get_spec()->get_column_attr(col_ndx);
+        if (!(attr & col_attr_Indexed))
+            return;
+
+        size_t tmp;
+        size_t* parent_col = df::record_subdesc_path(*subdesc->get(), &tmp, &tmp + 1);
 
+        if (!parent_col) {
+            // Adding or removing search index on a subtable of a subtable is not yet supported
+            throw LogicError(LogicError::subtable_of_subtable_index);
+        }
+
+
+        // Iterate through all rows of the root table and create an instance of each subtable. We have a special
+        // problem though: All subtables share the same common instance of attributes, however while we successively
+        // remove indexes, we have some subtables that will match the index flag and some that will not, which
+        // is an inchoherent state of the database. We rely on the fact that all method calls and operations in the 
+        // for-loop are safe to call despite of this.
+        for (size_t r = 0; r < size(); r++) {
+            // Set index bit from shared spec because we're now going to operate on the next subtable
+            // object which still has an index (because various method calls may crash if attributes are 
+            // wrong)
+            attr &= col_attr_Indexed;
```

Comment:
This seems wrong. All other attributes than `col_attr_Indexed` are removed. Maybe `|=` was intended?